spatialos-sdk/src/worker/parameters.rs

```diff
@@ -358,7 +358,7 @@ impl ProtocolLoggingParameters {
             CString::new(self.log_prefix.clone()).expect("Received 0 byte in supplied log prefix.");
 
         Worker_ProtocolLoggingParameters {
-            log_prefix: log_prefix_cstr.as_ptr(),
+            log_prefix: "protocol-log-".as_ptr() as _,//log_prefix_cstr.as_ptr(),
```

Comment:
As we discussed offline - we need to change this implementation

Reply:
Yep. Will fix this properly later.

Reply:
I've addressed the safety issue here by having `ProtocolLoggingParameters` own the `CString` directly. In order to avoid leaking implementation details (i.e. that it's using `CString` under the hood), I've made all the fields private and added setters. I've also added a note in the doc comment about allowing us to add parameters without breaking changes (similar to `CommandParameters`), though if that's not appropriate here let me know.

I've also opted to use `expect()` on the result of the `CString` conversion instead of returning a `Result<_, NulError>`. After giving it more thought, I don't think it makes a ton of sense to expose that error directly. While it's _technically_ possible for a Rust string to contain a nul byte in the middle, it's both incredibly unlikely to happen in practice, and it's likely indicative of an error in the user's code.